src/audio_unit/looper.rs

```diff
@@ -6,13 +6,77 @@ use State::*;
 #[derive(Debug)]
 pub enum Message {
     Toggle,
+    QueueOverdub,
 }
 
 #[derive(Debug)]
 enum State {
     Off,
     Recording { position: usize },
     Playing { position: usize, total: usize },
+    PlayingAwaitingOverdub { position: usize, total: usize },
+    Overdubbing { position: usize, total: usize },
+}
+
+impl State {
+    pub fn is_awaiting_overdub(&self) -> bool {
+        matches!(self, State::PlayingAwaitingOverdub { .. })
+    }
+
+    pub fn start_overdubbing(&mut self, position: usize) {
```

Comment:
Is this sort of what you had in mind re `impl State`? It sort of makes sense that these would be "declarative mutating methods" encapsulating the state transitions at certain moments in time?

Reply:
Yeah, that's what I meant. `start_overdubbing` feels a little weird since it's only valid in the one state, but I think I like the rest.

Reply:
Ya I initially tried to encapsulate the checking + setting in a single eg `start_overdubbing_if_queued()` method (which nicely avoided the conditionally-valid-invocation aspect) but introduced a bug in doing so (since the logic that invokes these state transitions is "if next_position == total and we're currently awaiting overdub, then start overdubbing, otherwise just update position", not sure what the conceptually elegant way would be to express that across the interface to this state transition logic)

So ya this doesn't feel completely resolved but seems fine for now, I'll go ahead and merge